CMakeLists.txt

```diff
@@ -824,6 +824,15 @@ else()
   set(ARCH "generic")
 endif()
 
+# Require SSE2 when targetting 32-bit x86.
+# SS2 is usually enabled by default for GCC and Clang, but not always.
+# See: https://github.com/aws/aws-lc/commit/6fe8dcbe96e580ea85233fdb98a142e42951b70b
+if(ARCH STREQUAL "x86" AND NOT OPENSSL_NO_SSE2_FOR_TESTING)
```

Comment:
Sorry, I wasn't aware that it was a build option. I think it's fine to have it then.

Reply:
Note: MSVC has a similar flag `/arch:SSE2` -- but I'm not sure it's ever needed since it the default for MSVC.

https://learn.microsoft.com/en-us/cpp/build/reference/arch-x86?view=msvc-170

Reply:
Yeah, I went through the same thinking that it's not needed since it's the default. Wasn't sure if it was too much information to add that in the comments. I tried to sum it up with "Most compilers enable SSE2 in 32-bit x86 by default, but in some cases GCC and Clang don't."